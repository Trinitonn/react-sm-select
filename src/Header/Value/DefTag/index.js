import React from 'react';

export const DefTag = ({label, index, removableTag, onTagRemove}) => (
  <div className="Header__tag">
    <div className="Header__tag__label">{label}</div>
    {removableTag && <div className="Header__tag__close" onClick={event => onTagRemove(index, event)}>âœ•</div>}
  </div>
);
