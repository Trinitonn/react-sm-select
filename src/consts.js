export const MODE = {
  LIST: 'list',
  TAGS: 'tags',
  COUNTER: 'counter',
  SINGLE: 'single',
};
