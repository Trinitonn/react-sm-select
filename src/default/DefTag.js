import React from 'react';

export const DefTag = ({label, index, removableTag, onTagRemove}) => (
  <div className="MultiSelect__tag">
    <div className="MultiSelect__tag__label">{label}</div>
    {removableTag && <div className="MultiSelect__tag__close" onClick={event => onTagRemove(index, event)}>âœ•</div>}
  </div>
);
