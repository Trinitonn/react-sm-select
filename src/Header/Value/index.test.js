import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { MODE } from '../../consts';

import { Value } from './index';
import { DefTag } from './DefTag';

const props = {
  mode: MODE.LIST,
  value: ['one'],
  options: [
    { value: 'one', label: 'One' },
    { value: 'two', label: 'Two' },
    { value: 'five', label: 'Five' },
  ],
  valuePlaceholder: 'Value Placeholder',
  allSelectedLabel: 'All are Selected',
  counterLabel: 'Counter Label',
  removableTag: true,
  onRemove: jest.fn(),
};

describe('Value', () => {
  it('should render properly default value', () => {
    const wrapper = shallow(<Value {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly custom value', () => {
    const wrapper = shallow(<Value {...props} Value={() => (<div />)}/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly all selected', () => {
    const wrapper = shallow(<Value {...props} value={['one', 'two', 'five']}/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Tags', () => {
    const wrapper = shallow(<Value {...props} mode={MODE.TAGS} value={['one', 'two', 'five']}/>);
    wrapper.find(DefTag).at(0).prop('onTagRemove')();

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(props.onRemove).toHaveBeenCalled();
  });

  it('should render Custom Tags', () => {
    const wrapper = shallow(
      <Value {...props} mode={MODE.TAGS} value={['one', 'two', 'five']} Tag={() => (<div />)}/>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Counter', () => {
    const wrapper = shallow(<Value {...props} mode={MODE.COUNTER} value={['one', 'two', 'five']}/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
