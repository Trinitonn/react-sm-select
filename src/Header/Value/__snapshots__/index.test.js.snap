// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Value should render Counter 1`] = `
<Counter
  counterLabel="Counter Label"
  options={
    Array [
      Object {
        "label": "One",
        "value": "one",
      },
      Object {
        "label": "Two",
        "value": "two",
      },
      Object {
        "label": "Five",
        "value": "five",
      },
    ]
  }
  value={
    Array [
      "one",
      "two",
      "five",
    ]
  }
  valuePlaceholder="Value Placeholder"
/>
`;

exports[`Value should render Custom Tags 1`] = `
<div
  className="Header__tags"
>
  <Tag
    index={0}
    key="0"
    label="One"
    onTagRemove={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    removableTag={true}
  />
  <Tag
    index={1}
    key="1"
    label="Two"
    onTagRemove={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    removableTag={true}
  />
  <Tag
    index={2}
    key="2"
    label="Five"
    onTagRemove={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    removableTag={true}
  />
</div>
`;

exports[`Value should render Tags 1`] = `
<div
  className="Header__tags"
>
  <DefTag
    index={0}
    key="0"
    label="One"
    onTagRemove={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    removableTag={true}
  />
  <DefTag
    index={1}
    key="1"
    label="Two"
    onTagRemove={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    removableTag={true}
  />
  <DefTag
    index={2}
    key="2"
    label="Five"
    onTagRemove={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    removableTag={true}
  />
</div>
`;

exports[`Value should render properly all selected 1`] = `
<span>
  All are Selected
</span>
`;

exports[`Value should render properly all selected with custom Value 1`] = `
<Numbers
  options={
    Array [
      Object {
        "label": "One",
        "value": "one",
      },
      Object {
        "label": "Two",
        "value": "two",
      },
      Object {
        "label": "Five",
        "value": "five",
      },
    ]
  }
  value={
    Array [
      "one",
      "two",
      "five",
    ]
  }
/>
`;

exports[`Value should render properly custom value 1`] = `
<Value
  options={
    Array [
      Object {
        "label": "One",
        "value": "one",
      },
      Object {
        "label": "Two",
        "value": "two",
      },
      Object {
        "label": "Five",
        "value": "five",
      },
    ]
  }
  value={
    Array [
      "one",
    ]
  }
/>
`;

exports[`Value should render properly default value 1`] = `
<DefValue
  options={
    Array [
      Object {
        "label": "One",
        "value": "one",
      },
      Object {
        "label": "Two",
        "value": "two",
      },
      Object {
        "label": "Five",
        "value": "five",
      },
    ]
  }
  value={
    Array [
      "one",
    ]
  }
/>
`;
